syntax = "proto3";

package models;

message Profile {
  int32 id = 1;
  int32 user_id = 2;
  string avatar_url = 3;
  string nickname = 4;
  string bio = 5;
  string additional_info = 6; // Assuming JSON serialized as string
}

service ProfileService {
  rpc CreateProfile (CreateProfileRequest) returns (Profile);
  rpc UpdateProfile (UpdateProfileRequest) returns (Profile);
  rpc GetProfileById (GetProfileByIdRequest) returns (Profile);
  rpc GetProfileByUserId (GetProfileByUserIdRequest) returns (Profile);
  rpc GetAllProfiles (GetAllProfilesRequest) returns (GetAllProfilesResponse);

}

message CreateProfileRequest {
  int32 user_id = 1;
  string avatar_url = 2;
  string nickname = 3;
  string bio = 4;
  string additional_info = 5; // JSON serialized as string
}

message UpdateProfileRequest {
  int32 id = 1;
  int32 user_id = 2;
  string avatar_url = 3;
  string nickname = 4;
  string bio = 5;
  string additional_info = 6; // JSON serialized as string
}

message GetProfileByIdRequest {
  int32 id = 1;
}

message GetProfileByUserIdRequest {
  int32 user_id = 1;
}
   message GetAllProfilesRequest {
     int32 limit = 1;  // Number of profiles per page
     int32 offset = 2; // Offset for pagination
   }

   message GetAllProfilesResponse {
     repeated Profile profiles = 1; // List of profiles
     int32 total_count = 2;         // Total number of profiles (for pagination metadata)
   }
